### ISSUE TITLE
Analyzing Data with Aggregation

### ISSUE THREAD

**Issue by sarukondadivya** [andres-fr/mongodb-coursera#2]
Go ahead and download the notebook attached and open it in Jupyter: analyzing-data-with-aggregation.

Every Citibike trip has a start and end station. In the trips collection, each document has a start station id and an end station id representing these stations.

Your task is to build an aggregation pipeline in cell 4 of the notebook to answer the following question: **Citibike trips that start at station id 279 end most frequently at what station id?**

import pymongo
import pprint
client = pymongo.MongoClient('mongodb://analytics-student:analytics-password@cluster0-shard-00-00-jxeqq.mongodb.net:27017,cluster0-shard-00-01-jxeqq.mongodb.net:27017,cluster0-shard-00-02-jxeqq.mongodb.net:27017/?ssl=true&replicaSet=Cluster0-shard-0&authSource=admin')
trips = client.citibike.trips
# Replace XXXX with your aggregation pipeline to answer the question:
# Citibike trips that start at station id 279 end most frequently at what station id?
pipeline = XXXX
# print the results
for trip in trips.aggregate(pipeline):
    pprint.pprint(trip)

**Comment by vasanth-6120** [comment]
Answer for this please

**Comment by vasanth-6120** [comment]
To verify that you've successfully completed this exercise enter your answer below.


what should I give as answer?

**Comment by vasanth-6120** [comment]
Answer please

**Comment by AutomateWithGautam** [comment]
analyzing-data-with-aggregation.
```
pipeline = [
    {
        '$match': {'start station id': 279, 'end station id': 279}
    },
    {
        '$sortByCount': "$end station id"
    },
    {
        '$limit': 1
    }
]
```
answer will be 279

**Comment by artodto** [comment]
hi!
These are some data that I have, <<373, 441, 293, 18414>>, but I don't know if they are true.
is only that answer please!
  could you help me please!

**Comment by artodto** [comment]
> analyzing-data-with-aggregation.
> 
> ```
> pipeline = [
>     {
>         '$match': {'start station id': 279, 'end station id': 279}
>     },
>     {
>         '$sortByCount': "$end station id"
>     },
>     {
>         '$limit': 1
>     }
> ]
> ```
> 
> answer will be 279

hi!
¿the answer is 279?

**Comment by alittleton98** [comment]
That answer is wrong. It assumes the end station id is 279, but the question specifically asks WHICH end station do trips starting from 279 end at the most. specifying an end station breaks the logic. Also the answer from this code is not 279 nor is it the actual answer.

**Comment by svonhatten** [comment]
Yes, I agree with @alittleton98 . The code answer provided is assuming the ending station id we should be looking at only is end station id 279. Removing that part from the `'$match': {'start station id': 279, 'end station id': 279}` to instead make that line be `'$match': {'start station id': 279)` only, and keep the `'$sortByCount'` and `'$limit'` code sections in the pipeline as well. That should give you the id 259 instead as your answer.

**Comment by Meghaahirwar98** [comment]
this is good things for everyone

**Comment by M0hc3n** [comment]
Personally I did this solution but I am not really sure whether it is true or not
`pipeline = [
    {
        '$match': {'start station id': 279}
    },
    {
        '$sort': { 'end station id' : -1}
    },
    {
        '$limit': 1
    }
]`

All what I am trying to do is grabing the trips that their start station id is 279 using '$match' then sorting them descendingly according to end station id and then setting limit to 1 because I only need the first element
